common/datavalues/src/types/data_type.rs

```diff
@@ -62,6 +64,9 @@ pub enum DataType {
     List(Box<DataField>),
     Struct(Vec<DataField>),
     String,
+
+    Enum8(BTreeMap<String, u8>),
```

Comment:
@sundy-li Is this definition OK?

Reply:
Yes, it's ok. I'm sorry that can you hold this pr after I finish #3794 ?

We will remove enum based `DataType` so there will be conflicts.

Reply:
Yes, I already notice that PR. But I want to understand how it works e2e first. I still cannot find the where to handle the insert statement transfer enum string to uint.

Reply:
Aha, it's related to sqlparser [https://github.com/datafuse-extras/sqlparser-rs] crate.